plugin/server_audit/server_audit.cc

```diff
@@ -1121,6 +1273,7 @@ static int start_logging()
     strncpy(current_log_buf, "[SYSLOG]", sizeof(current_log_buf)-1);
     compile_time_assert(sizeof current_log_buf > sizeof "[SYSLOG]");
   }
+  thr_timer_set_period(&audit_timer, log_buffer_time*1000);
```

Comment:
You do not really need periodic timer, as far as I can see  Periodic will tick, even if there is no work to do, this is a bit suboptimal. On the other hand, I think you know exactly when set that timer - if you the push first message to  queue, without flushing.